### DISCUSSION TITLE
About Marc Stammerjohann lecture in Prisma Day 2021 - Serializing Decimals

### DISCUSSION THREAD

**Discussion by DanielMaranhao** [prisma/prisma#13335] [Q&A] [answered]
Hello to all, sorry if this question is already answered, I didn't know how to search for this properly.

It is related to this lecture - https://www.youtube.com/watch?v=mmbd5hcQUaY

In it, there is a moment that Marc states: When you have a model in schema.prisma, if it has a Decimal property, it will be fetched as a string, because it uses the Decimal.js library. For example:

```
model Product {
  id        String   @id @default(cuid())
  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt

  name        String  @unique @db.VarChar(100)
  description String? @db.VarChar(500)
  price       Decimal @db.Decimal(6, 2)
  imgUrl      String

  categories Category[]
  items      OrderItem[]
}
```

To fetch this entity and receive the price property as a number, you have to do all that process explained in the video, which involves putting the result of the prisma query inside the instantiation of an entity.

So, instead of having just
`return this.productsService.findOne(id);`

You need to have
`return new ProductEntity(await this.productsService.findOne(id));`

The problem with this arises when I fetch an entity with relations. So in my case for example, I want to fetch orders, and bring together the associated items, which themselves have associated products that I also want to bring. In this case, I don't think there is a way to instantiate the relations, as I did with the entity itself that I fetched. So, if I leave the `ClassSerializerInterceptor` on, This decimal value won't even be brought as a string, but as this weird structure:

```
"price": {
	"s": 1,
	"e": 1,
	"d": [
		90,
		5000000
	]
},
```

So, does someone know what would be the correct course of action in this case? Thanks, and best wishes.

**Comment by pantharshit00** [comment, marked as answer]
Hello @DanielMaranhao , our apologies for the delay in getting back to you! Due to the large number of questions in our community channels, unfortunately this slipped through the cracks. 

If this is still relevant, can you please share the project in a Git repo? That way I can look up the code structure and try to suggest a solution. I looked in the workshop repository but there was no solution available so I am asking you here.

**Reply by DanielMaranhao** [reply]
Thanks for your reply, @pantharshit00. After some time studying and talking with colleagues, if I'm not mistaken, I noticed that it is something common of Node.js to return decimals from the database (in my case, postgres) as strings. Something about converting to string as to not lose precision, as returning as number would lose precision/decimal places, something like that. So I think it is not such a big problem, since probably the data will already return ready to use. But if I misunderstood something, feel free to correct where necessary. Thanks again and all the best wishes.

**Reply by pantharshit00** [reply]
Yes, you are correct here.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
